packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/TypeFactory.cs

```diff
@@ -117,14 +193,34 @@ public TypeProvider CreateEnum(InputEnumType enumType)
             return enumProvider;
         }
 
-        protected virtual TypeProvider CreateEnumCore(InputEnumType enumType) => EnumProvider.Create(enumType);
+        private TypeProvider? CreateEnumCore(InputEnumType enumType)
+        {
+            TypeProvider? type = null;
+            if (Visitors.Count == 0)
+            {
+                return EnumProvider.Create(enumType);
+            }
+            foreach (var visitor in Visitors)
+            {
+                type = visitor.Visit(enumType, type);
```

Comment:
shouldn't we pass in the type we would have done as a "base" implementation?  Also shouldn't these be refs so each change propagates to the next call if someone resets to null?